pom.xml

```diff
@@ -751,6 +747,171 @@
           </formats>
         </configuration>
       </plugin>
+
+      <plugin>
+        <groupId>org.jacoco</groupId>
+        <artifactId>jacoco-maven-plugin</artifactId>
+        <version>${maven.jacoco.plugin.version}</version>
+        <executions>
+          <execution>
+            <id>default-instrument</id>
+            <goals>
+              <goal>instrument</goal>
+            </goals>
+            <phase>test-compile</phase>
+          </execution>
+          <execution>
+            <id>default-restore-instrumented-classes</id>
+            <goals>
+              <goal>restore-instrumented-classes</goal>
+            </goals>
+          </execution>
+          <execution>
+            <id>default-report</id>
+            <phase>site</phase>
+            <goals>
+              <goal>report</goal>
+            </goals>
+          </execution>
+          <execution>
+            <id>default-check</id>
+            <goals>
+              <goal>check</goal>
+            </goals>
+            <configuration>
+              <dataFile>${project.build.directory}/jacoco/jacoco.exec</dataFile>
```

Comment:
let make variable for `${project.build.directory}/jacoco/jacoco.exec` to reuse it.

http://www.eclemma.org/jacoco/trunk/doc/report-mojo.html
>"Default value is: ${project.build.directory}/jacoco.exec."

default value for path is different, why to use non default ?

Reply:
I dont know. I took the existing commit and developed it a little.
Using the default value seems to solve a lot of problems.